File: webui/src/models/summoner.ts
import MatchSummary from "./matchSummary";

export default interface Summoner {
    id: string,
    accountId: string,
    puuid: string,
    name: string,
    profileIconId: number,
    revisionDate: number,
    summonerLevel: number,
    matchHistory: MatchSummary[]
};

File: API/internal/models/summoner.go
package models

type Summoner struct {
	Id string `json:"id"`
	AccountId string `json:"accountId"`
	Puuid string `json:"puuid"`
	Name string `json:"name"`
	ProfileIconId uint32 `json:"profileIconId"`
	RevisionDate uint64 `json:"revisionDate"`
	SummonerLevel uint32 `json:"summonerLevel"`
    MatchHistory []MatchSummary `json:"matchHistory"`
}
